datasets/castor_dataset.py

```diff
@@ -49,3 +50,21 @@ def __init__(self, path, load_ext_feats=False):
                 examples.append(example)
 
         super(CastorPairDataset, self).__init__(examples, fields)
+
+    @classmethod
+    def set_vectors(cls, field, vector_path):
+        if os.path.isfile(vector_path):
+            stoi, vectors, dim = torch.load(vector_path)
+            field.vocab.vectors = torch.Tensor(len(field.vocab), dim)
+
+            for i, token in enumerate(field.vocab.itos):
+                wv_index = stoi.get(token, None)
+                if wv_index is not None:
+                    field.vocab.vectors[i] = vectors[wv_index]
+                else:
+                    # initialize <unk> with uniform_(-0.05, 0.05) vectors
+                    field.vocab.vectors[i] = torch.FloatTensor(dim).uniform_(-0.05, 0.05)
+        else:
+            print("Error: Need word embedding pt file")
+            exit(1)
```

Comment:
You need to `from sys import exit` before you can do this I think.

Reply:
This is not imported in this file.

Reply:
https://stackoverflow.com/questions/28621112/if-i-import-exit-from-sys-which-exit-will-it-run
They are different but sys seems safer.